videosink/handler.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2021 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package videosink
+
+import (
+	"bytes"
+	"fmt"
+	"image/jpeg"
+	"log"
+	"mime"
+	"net/http"
+	"net/textproto"
+	"net/url"
+	"sync"
+)
+
+// bufferPool stores reusable []byte instances.
+var bufferPool = sync.Pool{
+	New: func() interface{} {
+		return []byte(nil)
+	},
+}
+
+type imageConfig struct {
+	format ImageFormat
+}
+
+func (d *Display) configFromQuery(values url.Values) (imageConfig, error) {
+	cfg := imageConfig{
+		format: d.defaultFormat,
+	}
+
+	if value := values.Get("format"); value != "" {
+		if format, err := ImageFormatFromString(value); err != nil {
+			return imageConfig{}, err
+		} else {
+			cfg.format = format
+		}
+	}
+
+	return cfg, nil
+}
+
+type client struct {
+	refresh   chan struct{}
+	terminate chan struct{}
+}
+
+func (d *Display) bufferChangedLocked() {
```

Comment:
I would prefer to have all methods in the same file where the struct is defined. Otherwise one has to scan through multiple files to discover all the private methods (public ones can be found via godoc).

Reply:
I agree with you and usually do the same. In this particular instance I elected to deviate and keep client-related code in `handler.go`. Well, mostly, because I just noticed that I added client-related code to `Display.Halt`.

Are you fine with the separation given the explanation or should I combine the code again?

Reply:
I don't feel the files are long enough to warrant splitting, but I don't feel strongly about that.

Reply:
If you don't mind I'll leave it split it up then. That way the per-file focus is retained and the tests match up too.